packages/components/src/DataList/DataList.types.ts

```diff
@@ -44,6 +44,11 @@ export type DataListHeader<T extends DataListObject> = {
   readonly [K in keyof T]?: string;
 };
 
+export interface DataListSorting {
+  readonly key: string;
```

Comment:
have you tried narrowing this down to whatever keys is available on the data/header?

Reply:
@rodrigoeidelvein , forgot to update this to be `Array<keyof DataListHeader<T>>` as well.

Also, maybe we should use `(keyof DataListHeader<T>)[]` here and on line 112 for consistency

Reply:
After Rodrigo's last comment. this actually makes sense as a string. Leaving it as is